torch/nn/_functions/thnn/upsampling.py

```diff
@@ -6,10 +6,10 @@
 from . import _all_functions
 
 
-class _UpsamplingBase(Function):
-
+class UpsamplingNearest2d(Function):
+    
     def __init__(self, size=None, scale_factor=None):
-        super(_UpsamplingBase, self).__init__()
+        super(UpsamplingNearest2d, self).__init__()
```

Comment:
Upsampling nearest restricts scaling to have equal aspect ratios and integer values [at the C level](https://github.com/torch/nn/blob/master/lib/THNN/generic/SpatialUpSamplingNearest.c#L8).  In contrast, bilinear sampling is more flexible and uses the specified [output size directly](https://github.com/torch/nn/blob/master/lib/THNN/generic/SpatialUpSamplingBilinear.c#L11-L12).  Perhaps there is a way to do NN with differing output aspect ratios, but that seems out of the scope of this PR or even this repo.